### ISSUE TITLE
Add a badge CSS style for tables

### ISSUE THREAD

**Issue by uhafner** [jenkinsci/design-library-plugin#42]
### What feature do you want to see added?

For table cells we currently have the styling 

```
jenkins-table__link jenkins-table__badge model-link inside
```

to show a badge-style link for model elements:

<img width="43" alt="Bildschirmfoto 2022-04-30 um 21 10 25" src="https://user-images.githubusercontent.com/503338/166119395-6794d0ff-c132-4d3a-bc6c-fd070def166d.png">

It would be helpful for plugins like the warnings and code coverage plugins (and maybe a lot of otheres), if there would be similar stylings available to show plugin results with a specific color. E.g. the code coverage as a percentage, where the background color can be customized by the plugin, e.g. red on bad coverage results, green for good, etc.

<img width="78" alt="Bildschirmfoto 2022-04-30 um 21 12 56" src="https://user-images.githubusercontent.com/503338/166119517-ab558878-0271-4c46-94c1-68ac0af5744f.png">

I tried to extract that part of the CSS from our Jenkins styles but I must admit that I did not understand how you achieved this existing styling in our complex CSS and LESS structure. Additionally, it might make sense if we would provide such kind of styling in a general way in Jenkins, so other plugins would not reinvent the wheel.

**Comment by timja** [comment]
cc @janfaracik

**Comment by janfaracik** [comment]
Good idea! :) 

I'm working on a just that for the eventual [plugins manager redesign](https://github.com/jenkinsci/jenkins/compare/master...janfaracik:new-plugin-manager-2-adopt-me?expand=1) as demoed way back in the day on the UX sig. 

It'll be as simple as adding a colour class e.g. `jenkins-!-color-pink` to change the tag colour.

**Comment by janfaracik** [comment]
I've opened a PR for this https://github.com/jenkinsci/jenkins/pull/6538 :)

**Comment by uhafner** [comment]
Ok thanks! If I understand correctly, https://github.com/jenkinsci/jenkins/pull/6538 only adds the colors. That means the badge still is not available as a separate CSS class?

**Comment by janfaracik** [comment]
> Ok thanks! If I understand correctly, [jenkinsci/jenkins#6538](https://github.com/jenkinsci/jenkins/pull/6538) only adds the colors. That means the badge still is not available as a separate CSS class?

If I'm understanding correctly, you'd want the 'badge' component on a page, not inside a table? Currently the badge has to be inside a table, I can work on changing that if so 👍

**Comment by uhafner** [comment]
No, I actually want the badge in the table but I did only manage to get the badge (`jenkins-table__badge`) that has the additional space for the model links, or am I missing something?

**Comment by mPokornyETM** [comment]
Will welcome the badge as extra component as well

**Comment by uhafner** [comment]
A private implementation of this badge now landed in core, see https://github.com/jenkinsci/jenkins/pull/10351. It would be really appreciated if we can move such useful widgets to our public library so that others do not need to reinvent the wheel.
